src/AppProviders.tsx

```diff
@@ -37,7 +39,11 @@ export function AppProviders({ children }: ProvidersProps) {
                       <MarketDataProvider>
                         <TransactionsProvider>
                           <ModalProvider>
-                            <DefiManagerProvider>{children}</DefiManagerProvider>
+                            <DefiManagerProvider>
+                              <ErrorBoundary FallbackComponent={ErrorPage}>
```

Comment:
are we able to put this higher in the provider tree such that it catches fatal errors in the providers above?